app/liquid_tags/glitch_tag.rb

```diff
@@ -31,49 +37,70 @@ def render(_context)
 
   private
 
-  def valid_id?(input)
-    (input =~ ID_REGEXP)&.zero?
+  def parsed_input(input)
+    id, *options = input.split
+    stripped_id = id.delete("~") # remove possible preceeding tilde
+    match = pattern_match_for(stripped_id, REGEXP_OPTIONS)
+    raise StandardError, I18n.t("liquid_tags.glitch_tag.invalid_glitch_id") unless match
+
+    [get_slug(match), parse_options(options, match)]
   end
 
-  def parse_id(input)
-    id = input.split.first
-    id.sub!(TILDE_PREFIX_REGEXP, "")
-    raise StandardError, I18n.t("liquid_tags.glitch_tag.invalid_glitch_id") unless valid_id?(id)
+  def get_slug(match)
+    if match_has_named_capture_group?(match, "slug_subdomain")
+      match[:slug_subdomain]&.delete(".") if match[:slug_subdomain].present?
+    else
+      match[:slug]
+    end
+  end
 
-    id
+  def match_has_named_capture_group?(match, group_name)
+    match.names.include?(group_name)
   end
 
-  def valid_option(option)
-    option.match(OPTION_REGEXP)
+  def parse_options(options, match)
+    # 'app' and 'code' should cancel each other out
+    options -= %w[app code] if (options & %w[app code]) == %w[app code]
+
+    # check for file= in options or path= in params; fallback is file=index.html
+    file_option = options.detect { |option| option.start_with?("file=") }
+    options += ["file=#{path_within_params(match)}"] unless file_option
+
+    validated_options = options.select { |option| valid_option?(option) }
+    raise StandardError, I18n.t("liquid_tags.glitch_tag.invalid_options") if validated_options.empty?
+
+    build_options(validated_options)
+  end
+
+  def path_within_params(match)
```

Comment:
Okay, I think this `path_within_params` method could use a refactor. Let me explain what is happening:

If I say the following:
```ruby
path_match = pattern_match_for(match[:params], [PATH_REGEX]) if match[:params].present?
```
I get an error on `match[:params]` if that named capture group does not exist, in spite of the `if` check. So I created the `match_has_named_capture_group?` method to ensure that the named capture exists before performing checks on it.

Now, from the former implementation, the idea is to have a fallback `file=` value of `index.html`. So, as I check for the presence of either the `params` and `path` named-captures, I want to return the fallback `index.html` value if either of these captures do not exist.

Hope I've made sense!